src/en/reference-bundle.md

```diff
@@ -0,0 +1,346 @@
+
+## Bundle file properties and format
+
+Comments (lines that begin with a '#' character) are used to explain various
+parts of the file. In addition, any line that is not a comment should be
+considered as an example. If such a line is duplicated it is for the sole
+purpose of showing another example. Do not duplicate lines in a real bundle
+file.
+
+A bundle file has six top level elements:
+
+`description`  
+`series`  
+`tags`  
+`applications`  
+`machines`  
+`relations`
+
+Some of these are simple key:value pairs while others contain nested
+configuration.
+
+```no-highlight
+#
+# description:
+#
+# Provides a description for the bundle. Visible in the Charm Store only.
+#
+
+description: This is a test bundle.
+
+description: |
+
+  This description is long and has multiple lines. Use the vertical bar as
+  shown in this example.
+
+#
+# series:
+#
+# Sets the default series for all applications in the bundle. This also affects
+# machines devoid of applications. See 'Charm series' above for how a final
+# series is determined.
+#
+
+series: bionic
+
+#
+# tags:
+#
+# Sets descriptive tags. A tag is used for organisational purposes in the
+# Charm Store. See https://docs.jujucharms.com/authors-charm-metadata for the
+# list of valid tags.
+#
+
+tags: [monitoring]
+
+tags: [database, utility]
+
+#
+# applications:
+#
+# The top level of a nested structure containing application data.
+#
+
+applications:
+
+  #
+  # <application name>:
+  #
+  # Sets the name of the application. It is typically the same as the charm
+  # name.
+  #
+  
+  easyrsa:
+
+    #
+    # charm:
+    #
+    # States what charm to use for the application. A fully qualified charm URI
+    # should be used for public bundles. In particular, the revision number
+    # should be appended to the charm name (e.g. name-123). A series can
+    # specified within the URI. See 'Charm series' above for how a final series
+    # is determined.
+    #
+
+    charm: cs:~containers/easyrsa-195
+
+    charm: cs:xenial/easyrsa-195
+
+    #
+    # series:
+    #
+    # Sets the series for the application. See 'Charm series' above for how a
+    # final series is determined.
+    #
+
+    series: bionic
+
+    #
+    # resources:
+    #
+    # States what charm resource to use. The keys are application-specific and
+    # are found within the corresponding charm's metadata.yaml file. To refer
+    # to the resource revision stored in the Charm Store an integer value is
+    # used. To refer to a local resource a string (file path) is used. For the
+    # latter, both absolute and relative (to the bundle file) paths are
+    # supported.
+    #
+
+    resources:
+      easyrsa: 5
+
+    resources:
+      easyrsa: ./relative/path/to/file
+
+    resources:
+      easyrsa: /absolute/path/to/file
+
+    #
+    # num_units:
+    #
+    # Specifies the number of units to deploy. The default value is '0'. 
+    #
+
+    num_units: 2
```

Comment:
Only relevant for non-k8s bundles. "scale" is used for k8s bundle.
num_units is supported for k8s but scale is recommended and preferred